hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/ReactiveQuery.java

```diff
@@ -121,4 +121,6 @@
 	ReactiveQuery<R> setCacheRegion(String cacheRegion);
 
 	String getCacheRegion();
+
+	ReactiveQuery<R> setQuerySpaces(String[] querySpaces);
```

Comment:
Shouldn't we expose this only on native queries?

Reply:
Um, in principle yes, but we don't have a `ReactiveNativeQuery` type at present. (Notice that this is an internal API, not one that is visible to users.)

Reply:
ha, we have some "soft conventions" about what's API. Pretty much anything which isn't explicitly hidden in a "spi" or "impl"/"internal" package is being treated as API in other projects.

That's just how it is on other projects; I don't mean to impose any cargo-cult here, but that made me assume this is API and I suspect it will give the same impression to other users.

Do you think we could improve on this, or not worth it?  cc/ @DavideD

Reply:
I think for the most part that's what we are doing already. We might need to give it another review before releasing a CR.

Reply:
@Sanne We have API packages and internal packages. I've been putting the APIs of internal packages in the package root, and the implementation in `.impl`. Objectively there's not a lot of difference just because we don't have non-public packages (yet) in Java.

Reply:
We do have non-public packages if you embrace the module system.

Reply:
> We do have non-public packages if you embrace the module system.

Yes sure, I should have said "in the version of Java we're stuck on for now".

Reply:
@Sanne I've met you halfway. Check it out.